burr/core/application.py

```diff
@@ -77,6 +77,19 @@ def _validate_result(result: dict, name: str) -> None:
         )
 
 
+def _adjust_single_step_output(output: Union[State, Tuple[dict, State]]):
+    """Adjusts the output of a single step action to be a tuple of (result, state)"""
+    if isinstance(output, tuple):
```

Comment:
this should inspect that it's:

1. two items
2. the first is a dict, the second is State

Reply:
and not the other way around, etc.